spvc/test/run_spirv_cross_tests.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2019 The Shaderc Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+Run the spirv-cross tests on spvc.
+
+Repeat runs are fast because we use a saved, compiled version of each shader
+instead of re-compiling.  To re-compile the shaders just delete the cache.
+'''
+
+from __future__ import print_function
+
+import errno
+import filecmp
+import os
+import subprocess
+import sys
+import tempfile
+
+test_case_dirs = (
+# directory           language  optimize
+('shaders'            , 'glsl', ''   ),
+('shaders'            , 'glsl', 'opt'),
+('shaders-no-opt'     , 'glsl', ''   ),
+('shaders-msl'        , 'msl' , ''   ),
+('shaders-msl'        , 'msl' , 'opt'),
+('shaders-msl-no-opt' , 'msl' , ''   ),
+('shaders-hlsl'       , 'hlsl', ''   ),
+('shaders-hlsl'       , 'hlsl', 'opt'),
+('shaders-hlsl-no-opt', 'hlsl', ''   ),
+('shaders-reflection' , 'refl', ''   ),
+)
+
+test_count = 0
+pass_count = 0
+
+not_used, tmpfile = tempfile.mkstemp()
+devnull = open(os.devnull, 'w')
+
+def mkdir_p(filepath):
+    ''' make sure directory exists to contain given file '''
+    dirpath = os.path.dirname(filepath)
+    try:
+        os.makedirs(dirpath)
+    except OSError as exc:
+        if not (exc.errno == errno.EEXIST and os.path.isdir(dirpath)):
+            raise
+
+def spirv_as(inp, out):
+    subprocess.check_call([spirv_as_path, '-o', out, inp], stdout=devnull)
```

Comment:
Do we still get stderr if the compile fails?

Reply:
yes